src/main/java/jenkinsci/plugins/influxdb/InfluxDbPublisher.java

```diff
@@ -215,53 +215,59 @@ public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, Task
             addPoints(pointsToWrite, cdmGen, listener);
         }
 
-        CoberturaPointGenerator cGen = new CoberturaPointGenerator(measurementRenderer, customPrefix, build);
-        if (cGen.hasReport()) {
-            listener.getLogger().println("[InfluxDB Plugin] Cobertura data found. Writing to InfluxDB...");
-            addPoints(pointsToWrite, cGen, listener);
-        }
+        try {
+            CoberturaPointGenerator cGen = new CoberturaPointGenerator(measurementRenderer, customPrefix, build);
+            if (cGen.hasReport()) {
+                listener.getLogger().println("[InfluxDB Plugin] Cobertura data found. Writing to InfluxDB...");
+                addPoints(pointsToWrite, cGen, listener);
+            }
+        } catch (NoClassDefFoundError ignore) { }
 
         RobotFrameworkPointGenerator rfGen = new RobotFrameworkPointGenerator(measurementRenderer, customPrefix, build);
```

Comment:
@letsrokk is there any reason you skipped the Robot Framework with these changes? We're currently in a situation where we do not have the Robot Framework installed (we're trying to use the JUnit hookup) and we're getting:
```
ERROR: Step ‘Publish build data to InfluxDb target’ aborted due to exception: 
15:32:08 java.lang.NoClassDefFoundError: hudson/plugins/robot/RobotBuildAction
15:32:08 	at jenkinsci.plugins.influxdb.generators.RobotFrameworkPointGenerator.hasReport(RobotFrameworkPointGenerator.java:47)
15:32:08 	at jenkinsci.plugins.influxdb.InfluxDbPublisher.perform(InfluxDbPublisher.java:229)
15:32:08 	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
15:32:08 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
15:32:08 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
15:32:08 	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:720)
15:32:08 	at hudson.model.Build$BuildExecution.cleanUp(Build.java:195)
15:32:08 	at hudson.model.Run.execute(Run.java:1775)
15:32:08 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
15:32:08 	at hudson.model.ResourceController.execute(ResourceController.java:98)
15:32:08 	at hudson.model.Executor.run(Executor.java:404)
```

Reply:
@chiefy The reason I merged this without checking that was an honest mistake. The issue is noticed (https://issues.jenkins-ci.org/browse/JENKINS-49308) and fixed and will be available in the next release, probably still during this week.

Reply:
@chiefy no real reason apart from the fact that I didn't notice same behaviour because Robot Framework plugin was installed

@asimell sorry I didn't go through all plugins

Reply:
@asimell @letsrokk Thanks - just wanted to make sure I wasn't missing something. For now we can just install the Robot Framework plugin and hopefully it will be a workaround.